### ISSUE TITLE
(2.8) ObjectMapper._readMapAndClose() swallows IOException on close()

### ISSUE THREAD

**Issue by cowwoc** [FasterXML/jackson-databind#629]
`ObjectMapper._readMapAndClose()` contains this code:

```
            try {
                jp.close();
            } catch (IOException ioe) { }
```

Does it really make sense to do this? I know that exceptions are unlikely to be thrown when closing a file that was only read file (there is no data to flush on `close()`) but the method already declares it throws `IOException` so what's the harm of letting the exception bubble up?

**Comment by cowtowncoder** [comment]
Since Jackson 2.7 allows use of Java 7 types (although has to be dynamic wrt runtime), and 2.8 will allow regular use of language features, I think this can finally be done in 2.8. Will edit title to reflect that.

**Comment by cowtowncoder** [comment]
Fixed via #1221, for 2.8.0. `IOException` from close is not swallowed if it's the primary problem; if it's secondary, it will be linked as `suppressed` for primary exception so it is still retained.
